worker/caasoperator/caasoperator.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caasoperator
+
+import (
+	"path/filepath"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/clock"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/agent/tools"
+	jworker "github.com/juju/juju/worker"
+	"github.com/juju/juju/worker/caasoperator/commands"
+	"github.com/juju/juju/worker/catacomb"
+)
+
+var logger = loggo.GetLogger("juju.worker.caasoperator")
+
+// A CaasOperatorExecutionObserver gets the appropriate methods called when a hook
+// is executed and either succeeds or fails.  Missing hooks don't get reported
+// in this way.
+type CaasOperatorExecutionObserver interface {
+	HookCompleted(hookName string)
+	HookFailed(hookName string)
+}
+
+// CaasOperator implements the capabilities of the caasoperator agent. It is not intended to
```

Comment:
None of this is tested, and it doesn't do anything useful yet anyway. Can you please pull this worker into another PR, with tests for the bits that are there? It doesn't need to be useful, but adding tests as we go means we don't end up piling on top of potentially shaky foundations.

Reply:
It's tested in the jujud agent tests where we check that the symlinks are created. So there is test coverage for the bit that is implemented. We will add unit tests as the worker is fleshed out to do more than just start the loop and wait for destruction. The idea of this PR is to add the foundation on which to build up new stuff.